### PR TITLE
Added test for conversions.

### PR DESCRIPTION
**Pull request by luca-heltai**
@FabCreda , @fdrmrc 

- [x] to_moonolith per spacedim 3.
- [x] test per triangoli e quadrilateri (DimNoOneTester...)
- [x] implementazione per tetra, wedge, pyramid, hex
- [x] implementazione di compute_intersections using moonolith

### CODE REVIEW

File: include/deal_moon_conversion.hpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#  include <cassert>
+#  include <cmath>
+#  include <fstream>
+#  include <iostream>
+#  include <sstream>
+
+using namespace dealii;
+using namespace moonolith;
+
+
```
Comment (luca-heltai) [inline include/deal_moon_conversion.hpp:30]: can we enclose this in the moonolith namespace, i.e., `namespace moonolith { ... }`?

Comment (luca-heltai) [reply include/deal_moon_conversion.hpp:30]: Also, @FabCreda  has added the implementation of all the other functions. @FabCreda, can you move all function declarations from the test to the new file include file, and remove them from the test, while @fdrmrc implements the deal.II `compute_intersection` function?

**Comment by fdrmrc** [comment]
https://github.com/luca-heltai/fsi-suite/blob/96b122e8a8d479b79d5c37900f6629a9be26ce23/gtests/moonlith_interface.cc#L62
` const auto                      points  = q.get_points; `
https://github.com/luca-heltai/fsi-suite/blob/96b122e8a8d479b79d5c37900f6629a9be26ce23/gtests/moonlith_interface.cc#L63
` const auto                      points  = q.weights; `

I'll add a test for this

**Comment by luca-heltai** [comment]
I've made some changes, but things don't seem to work in 1d/1d and 1d/3d. @fdrmrc, can you try to fix this? One easy solution would be to explicitly create the missing functions and throw exceptions when they are used. Also, my linker complains about multiply defined functions...

**Pull request merged**